build/common.ps1

```diff
@@ -189,12 +189,36 @@ Function Install-NuGet {
     & $NuGetExe locals all -list -verbosity detailed
 }
 
+Function Get-MsBuildPath{
+    [CmdletBinding()]
+    param()
+
+    Push-Location $PSScriptRoot
+
+    try {
+        $path = .\vswhere.exe -latest -products * -requires Microsoft.Component.MSBuild -property installationPath
```

Comment:
vswhere is not added part of the PATH when VS is installed. 

You'd need to use the full location of the exe for the script to work universally. 

Not 100% where all this msbuild path is used, but I don't know if it's a good idea to default to the latest version of MSBuild we can find on the machine.

Reply:
oh I just noticed you added the binary in repo. That doesn't belong there.

Reply:
if it doesnt belong there, _vswhere_ does it need to go?   

That is more or less a copy of the [example](https://github.com/Microsoft/vswhere/wiki/Find-MSBuild), and the problem with the existing script is that its hardcoded to use the `c:\program files(x86)\...` path, and that is not where I have a vs installation at, so the script fails for anyone not using that default location.   `vswhere` is supposed to be the way to locate vs instances since the registry is no longer reliable/used. 

Utility binaries go in repos all the time. Its not evil when its needed.

Reply:
I did add the files to the SLN as those kind of artifacts are part of the solution and the code that must be maintained together  If there is a reason aside from a personal preference why they should not be present, please share. 

VS added the proj file changes. I didn't initiate those. It does that kind of often, and it should stop creating that kind of version control churn.

Reply:
as far as I am aware, vswhere.exe will always be located in "C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" 
I don't think you can change the location of the installer. 

So you can probably just hardcode that. 

As far as binaries in the repo, that's a no because it drastically increases the size of the repo forever, and it's a good rule of thumb to acquire tools like that rather than package them in. 

Basically cleanup the VS added fluff and just depend on the local version of vswhere and it should be a lot better.

Reply:
The wiki page that StingyJack linked says
> With Visual Studio 2017 Update 2 or newer installed, you can find vswhere at %ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe

VS release notes say that 15.0.2 was released in March 2017, and 15.2 was released in May 2017, so either way anyone who has updated/installed in the last 17 months should have it at that well defined location.

The linked docs also say that vswhere is available as a nuget package, so the powershell script could check for vswhere at the well known location, and if it's it's not there, then download it with nuget.exe. Both configure.ps1 and build.ps1 run install-nuget, so as long as Get-MsBuildPath is called after Install-NuGet, we know nuget.exe exists at .nuget\nuget.exe. Or just add vswhere to .nuget\packages.config and then we know that vswhere will always be installed to the packages folder and don't even try the VS Installer path.

Reply:
> VS release notes say that 15.0.2 was released in March 2017, and 15.2 was released in May 2017, so either way anyone who has updated/installed in the last 17 months should have it at that well defined location.

Based on my observations and experiences, this makes an assumption that VS users regularly take the updates issued. If I dont see anything useful in the release notes, I wont update sometimes for months. There are a few people that I work with and know that have not taken a VS 15 update for a much longer time primarily because they do not want to risk destabilizing their development environment but also for other reasons. With the recent announcement of 15.9 becoming "service pack 1", and all the interim versions between it and RTM going "unsupported", they will probably soon have to take an even larger risk. 

This isn't a dispute of your suggestions as realistically none of them are likely to ever try to run this script, or even be curious about the repo at all (or even know GH exists), but its important to remember these are your "moderately frequent adopters" and they do exist in the userbase, and its not abnormal for them to go 1-3 years between taking even minor updates. For them, this speed "feels" normal and sometimes even too fast relative to the infrequent adopters they tend to work with (end customers in healthcare and manufacturing will run the same software for 10-20 years or more). 



> Basically cleanup the VS added fluff and just depend on the local version of vswhere and it should be a lot better.

I cant guarantee that VS wont re-add the fluff,  but I'll try to get to the suggested updates over the weekend.